app/_components/QueryByGeneUtil.js

```diff
@@ -24,76 +24,54 @@ window.QueryByGeneUtil = (function() {
-      };
+        cancer_study_list: cohortIds,
+        cancer_study_id:'all'
+      }
+
+      if(geneIds !== undefined && geneIds !== ''){
+        formOps['tab_index'] = 'tab_visualize';
+        formOps['Action'] = 'Submit';
+        formOps['data_priority'] = 0;
+        formOps['gene_list'] = encodeURIComponent(geneIds);
+
+        var physicalStudies = _.pluck(stats.studies,'id')
+
+        if(cohortIds.length === 1 && physicalStudies.length === 1 && physicalStudies[0] === cohortIds[0]){
+          //TODO: what if window.mutationProfileId is null
+          formOps['genetic_profile_ids_PROFILE_MUTATION_EXTENDED'] = window.mutationProfileId;
+          //TODO: what if window.cnaProfileId is null
+          formOps['genetic_profile_ids_PROFILE_COPY_NUMBER_ALTERATION'] = window.cnaProfileId;
+          formOps['case_set_id'] = cohortIds[0]+'_all'
+        } else {
+          formOps['case_set_id'] = 'all'
+        }
+      }
 
-      if (_.isObject(selectedCases)) {
-        _.each(selectedCases, function(_obj) {
-          var _studyId = _obj.id;
-          _.each(_obj.samples, function(_sampleId) {
-            _arr.push(_studyId + ":" + _sampleId);
+      //check if there are filters
+      if ((JSON.stringify(stats.filters) !== JSON.stringify({patients:{},samples:{}})) || includeCases) {
+        var studySamples = [];
+        _.each(stats.studies, function(study) {
+          _.each(study.samples, function(sampleId) {
+            studySamples.push(study.id + ":" + sampleId);
           });
         });
-
-        formOps = {
-          cancer_study_list: selectedCases.map(function(t) {
-            return t.id
-          }),
-          cancer_study_id: 'all',
-          case_set_id: -1,
-          case_ids: _arr.join('+')
-        };
+        formOps['case_set_id'] = -1;
+        formOps['case_ids'] = studySamples.join('+');
+      }
+      if(includeCases){
+        formOps['cancer_study_list'] = _.pluck(stats.studies,'id')
       }
       submitForm(window.cbioURL + 'index.do', formOps);
-    },
-    toQueryPageSingleCohort: function(studyId, selectedCases,
-                                      selectedGenes, mutationProfileId, cnaProfileId) {
-      var _arr = [];
-      _.each(selectedCases, function(_obj) {
-        var _studyId = _obj.id;
-        _.each(_obj.samples, function(_sampleId) {
-          _arr.push(_studyId + ":" + _sampleId);
-        });
-      });
-      submitForm(window.cbioURL + 'index.do', {
-        cancer_study_list: selectedCases.map(function(x) {
-          return x.id
-        }),
-        cancer_study_id: 'all',
-        case_ids: _arr.join('+'),
-        case_set_id: -1,
-        gene_set_choice: 'user-defined-list',
-        gene_list: encodeURIComponent(selectedGenes),
-        Z_SCORE_THRESHOLD: 2.0,
-        genetic_profile_ids_PROFILE_MUTATION_EXTENDED: mutationProfileId,
-        genetic_profile_ids_PROFILE_COPY_NUMBER_ALTERATION: cnaProfileId,
-        clinical_param_selection: null,
```

Comment:
This is gone?

Reply:
I think this parameter is never used in back-end or front-end